src/main/java/arrow/RaiseAddImport.java

```diff
@@ -0,0 +1,55 @@
+package arrow;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.kotlin.KotlinIsoVisitor;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RaiseAddImport extends Recipe {
+    @JsonCreator
+    public RaiseAddImport() {
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Raise Rewrite";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Rewrites all imports, and builders from arrow.core.computations.* and arrow.core.continuations.* to arrow.core.raise.*.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new RaiseImportVisitor();
+    }
+
```

Comment:
```suggestion
    @Override
    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
        return Applicability.or(
                new UsesType<>("arrow.core.continuations.EffectScope"),
                new UsesType<>("arrow.core.continuations.EagerEffectScope")
        );
    }
```

Reply:
I'd add an applicability test here, which will pre-filter the source files on which the recipe will run.
`UsesType` will check if the source contains the type ahead of time to prevent unnecessary work.

Note: The checked types will depend on where this recipe is called: before ChangeType or after ChangeType.